optimum/onnxruntime/modeling_decoder.py

```diff
@@ -465,121 +484,214 @@ def _from_pretrained(
+            model_save_dir = new_model_save_dir
+
         return cls(
+            model[0],
             config,
-            *model,
+            decoder_with_past_session=model[1],
             use_io_binding=use_io_binding,
             model_save_dir=model_save_dir,
-            last_decoder_model_name=file_names["last_decoder_model_name"],
-            last_decoder_with_past_model_name=file_names.get("last_decoder_with_past_model_name", None),
+            preprocessors=preprocessors,
         )
 
     @classmethod
     def _from_transformers(
         cls,
         model_id: str,
         config: "PretrainedConfig",
-        subfolder: Optional[str] = "",
-        save_dir: Union[str, Path] = default_cache_path,
         use_auth_token: Optional[Union[bool, str]] = None,
-        revision: Optional[str] = None,
+        revision: str = "main",
         force_download: bool = True,
         cache_dir: Optional[str] = None,
+        subfolder: str = "",
+        local_files_only: bool = False,
         use_cache: bool = True,
-        **kwargs,
-    ):
-        # Create local save dir in cache dir
-        save_dir = Path(save_dir).joinpath(model_id)
-        save_dir.mkdir(parents=True, exist_ok=True)
-        preprocessor = get_preprocessor(model_id)
-        framework = FeaturesManager.determine_framework(os.path.join(model_id, subfolder))
-        model_class = FeaturesManager.get_model_class_for_feature(cls.export_feature, framework)
-        model = model_class.from_pretrained(model_id, subfolder=subfolder, config=config, cache_dir=cache_dir)
-
-        # Export the decoder without the past key values
-        onnx_config = DecoderOnnxConfigWithPast(model.config, task=cls.export_feature, use_past=False)
-        onnx_opset = onnx_config.default_onnx_opset
+        provider: str = "CPUExecutionProvider",
+        session_options: Optional[onnxruntime.SessionOptions] = None,
+        provider_options: Optional[Dict[str, Any]] = None,
+        use_io_binding: bool = True,
+        task: Optional[str] = None,
+    ) -> "ORTModelDecoder":
+        if task is None:
+            task = cls._AUTOMODELS_TO_TASKS[cls.auto_model_class]
+
+        save_dir = TemporaryDirectory()
+        save_dir_path = Path(save_dir.name)
+
+        model = TasksManager.get_model_from_task(
+            task,
+            model_id,
+            subfolder=subfolder,
+            revision=revision,
+            cache_dir=cache_dir,
+            config=config,
+            use_auth_token=use_auth_token,
+            local_files_only=local_files_only,
+            force_download=force_download,
+        )
+
+        model_type = model.config.model_type.replace("_", "-")
+        model_name = getattr(model, "name", None)
+
+        onnx_config_constructor = TasksManager.get_exporter_config_constructor(
+            model_type, "onnx", task=task, model_name=model_name
+        )
+        onnx_config = onnx_config_constructor(model.config, use_present_in_outputs=True)
```

Comment:
I see this being repeated in multiple files. Would it make sense to have a helper function for these lines?

Reply:
I tried implementing it, but end up with circular imports!